File: src/main/java/com/waitring/waitring/entity/StoreKeyword.java
package com.waitring.waitring.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

/**
 * 가게_키워드 엔티티
 */

@Entity
@Getter
@ToString(exclude = "store")
@Builder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class StoreKeyword extends BaseEntity {

    @Id @GeneratedValue
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "store_id", updatable = false)
    @JsonIgnore
    private Store store; // 가게

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "keyword_id", updatable = false)
    @JsonIgnore
    private Keyword keyword; // 키워드
}

File: src/main/java/com/waitring/waitring/entity/StoreKeyword.kt
package com.waitring.waitring.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.*

/**
 * 가게_키워드 엔티티
 */
@Entity
class StoreKeyword(store: Store, keyword: Keyword) : BaseEntity() {
    @Id
    @GeneratedValue
    var id: Long? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "store_id", nullable = false, updatable = false)
    @JsonIgnore
    var store = store // 가게
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "keyword_id", nullable = false, updatable = false)
    var keyword = keyword // 키워드
        protected set

    override fun toString(): String {
        return "StoreKeyword(id=$id, store=$store, keyword=$keyword)"
    }
}
